### PR TITLE
Hide warning about "www folder not found" for widgets adapters

### PR DESCRIPTION
**Pull request by GermanBluefox**
Fix https://github.com/ioBroker/ioBroker.vis/issues/641

### CODE REVIEW

**Comment by Apollon77** [comment]
Hm ... da steht aber "INFO:" ... wo steht da warning? :-)

**Review by Apollon77 — DISMISSED** [review]

**Comment by Apollon77** [comment]
Ok, that rephrasing is good I think

File: packages/cli/src/lib/setup/setupUpload.ts [hunk 1]
```diff
@@ -530,7 +530,8 @@ export class Upload {
 
         if (!fs.existsSync(dir)) {
             // www folder have not all adapters. So show warning only for admin folder
-            (isAdmin || (cfg && cfg.common && cfg.common.onlyWWW)) &&
+            // widgets does not have www folder, but they onlyWWW flag
+            (isAdmin || (cfg && cfg.common && cfg.common.onlyWWW && !cfg.common.visWidgets)) &&
```
[the remark this answers is not in the document]
Comment (foxriver76) [reply packages/cli/src/lib/setup/setupUpload.ts:534]: It is not in schema, if it exists we need to add it

Comment (foxriver76) [inline packages/cli/src/lib/setup/setupUpload.ts:534]:
```suggestion
            (isAdmin || (cfg?.common?.onlyWWW && !cfg.common.visWidgets)) &&
```

Comment (Apollon77) [reply packages/cli/src/lib/setup/setupUpload.ts:534]: But I like the added comments in the original version honestly ....

Comment (foxriver76) [reply packages/cli/src/lib/setup/setupUpload.ts:534]: With only for admin version? yes lets keep it if its still true ;-)

[the remark this answers is not in the document]
Comment (foxriver76) [reply packages/cli/src/lib/setup/setupUpload.ts:534]: I can add it to schema later that day

File: packages/cli/src/lib/setup/setupUpload.ts [hunk 2]
```diff
@@ -530,7 +530,8 @@ export class Upload {
 
         if (!fs.existsSync(dir)) {
             // www folder have not all adapters. So show warning only for admin folder
-            (isAdmin || (cfg && cfg.common && cfg.common.onlyWWW)) &&
+            // widgets does not have www folder, but they onlyWWW flag
+            (isAdmin || (cfg?.common?.onlyWWW && !cfg?.common?.visWidgets)) &&
```
Comment (foxriver76) [inline packages/cli/src/lib/setup/setupUpload.ts:534]:
```suggestion
            (isAdmin || (cfg?.common?.onlyWWW && !cfg.common.visWidgets)) &&
```
Still prefer this one, `cfg.common` always exists on entering the `&&` and it's hard to read at least in my mind beacause basically a non existing `cfg` would evaluate to `true`, even if this cannot happen in this scenario it is weird to read for me.

File: packages/cli/src/lib/setup/setupUpload.ts [hunk 3]
```diff
@@ -530,7 +530,8 @@ export class Upload {
 
         if (!fs.existsSync(dir)) {
             // www folder have not all adapters. So show warning only for admin folder
-            (isAdmin || (cfg && cfg.common && cfg.common.onlyWWW)) &&
+            // widgets does not have www folder, but they onlyWWW flag
```
Comment (foxriver76) [inline packages/cli/src/lib/setup/setupUpload.ts:533]:
```suggestion
            // widgets do not have www folder, but they have onlyWWW flag
```

**Pull request merged**